l3kernel/l3luatex.dtx

```diff
@@ -370,9 +370,10 @@ local scan_int     = token.scan_int or token.scan_integer
 local scan_string  = token.scan_string
 local scan_keyword = token.scan_keyword
 local put_next     = token.put_next
+local token_create = token.create
 
-local true_tok     = token.create'prg_return_true:'
-local false_tok    = token.create'prg_return_false:'
+local true_tok     = token_create'prg_return_true:'
+local false_tok    = token_create'prg_return_false:'
 %    \end{macrocode}
```

Comment:
`true_tok` and `false_tok` seem unused.  Besides, don't they end up being undefined cs for the same reason as you explained for `use_none` and `use_i`?

Reply:
They are used in `l3token` and defined here since they were considered to be generally useful. They would be undefined, but `l3bootstrap` explicitly makes sure that they appear to avoid that.

Reply:
Should we have a wrapper like (untested code, and doesn't deal correctly with crazy category codes; it should `tex.print` some of it with the `-2` catcodetable)
```
token_create_safe = function(s)
  tex.runtoks(function()
    tex.sprint("\\edef\\__luatex_dummy:{\z
        \\expandafter\\noexpand\\csname" .. s .. "\\endcsname\z
    }\\endgroup")
  end)
  return token_create(s)
end
```

Reply:
I like the idea, but would probably implement it based on `token.set_char` or so to avoid having to deal with catcode issues.

Reply:
I don't understand how to avoid leaving the weird meaning in place on the TeX side when using `token.set_char`.  Is there a `token.set_undefined` or similar (I found nothing in the manual)?

Reply:
There isn't. The idea would be to first run `\begingroup`, then `token.set_char` and finally `\endgroup`. That's a bit of a pain, but since other approaches probably require groups too it seems reasonable.

Reply:
I added a `token_create_safe`  based on`set_char` and `let`.

Reply:
Just looked at the code for `token_create_safe`.  It's really neat!